src/main/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Extension
+public class EnvInjectEnvVarsContributor extends EnvironmentContributor {
+
+    @Override
+    public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        for (Object property : job.getProperties().values()) {
+            if (property instanceof EnvInjectJobProperty) {
+                EnvInjectJobProperty jobProperty = (EnvInjectJobProperty) property;
+                EnvInjectJobPropertyInfo jobPropertyInfo = jobProperty.getInfo();
+                Map<String, String> result = jobPropertyInfo.getPropertiesContentMap(new HashMap<String, String>());
+                if (result != null) {
+                    env.putAll(result);
```

Comment:
I doubt it will work as expected if a property depends on the value of another property. A test is required to demonstrate it